sys/linux/dev_bifrost.txt

```diff
@@ -0,0 +1,370 @@
+ioctl$KBASE_IOCTL_STICKY_RESOURCE_MAP(fd fd_bifrost, cmd const[KBASE_IOCTL_STICKY_RESOURCE_MAP], arg ptr[in, kbase_ioctl_sticky_resource_map])
+ioctl$KBASE_IOCTL_STICKY_RESOURCE_UNMAP(fd fd_bifrost, cmd const[KBASE_IOCTL_STICKY_RESOURCE_UNMAP], arg ptr[in, kbase_ioctl_sticky_resource_unmap])
+ioctl$KBASE_IOCTL_MEM_FIND_GPU_START_AND_OFFSET(fd fd_bifrost, cmd const[KBASE_IOCTL_MEM_FIND_GPU_START_AND_OFFSET], arg ptr[inout, kbase_ioctl_mem_find_gpu_start_and_offset])
+ioctl$KBASE_IOCTL_MEM_EXEC_INIT(fd fd_bifrost, cmd const[KBASE_IOCTL_MEM_EXEC_INIT], arg ptr[in, kbase_ioctl_mem_exec_init])
+ioctl$KBASE_IOCTL_GET_CPU_GPU_TIMEINFO(fd fd_bifrost, cmd const[KBASE_IOCTL_GET_CPU_GPU_TIMEINFO], arg ptr[inout, kbase_ioctl_get_cpu_gpu_timeinfo])
+
+ioctl$KBASE_HWCNT_READER_GET_HWVER(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_GET_HWVER], arg const[0])
+ioctl$KBASE_HWCNT_READER_GET_BUFFER_SIZE(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_GET_BUFFER_SIZE], arg const[0])
+ioctl$KBASE_HWCNT_READER_DUMP(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_DUMP], arg const[0])
+ioctl$KBASE_HWCNT_READER_CLEAR(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_CLEAR], arg const[0])
+ioctl$KBASE_HWCNT_READER_GET_BUFFER(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_GET_BUFFER], arg ptr[out, kbase_hwcnt_reader_metadata])
+ioctl$KBASE_HWCNT_READER_PUT_BUFFER(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_PUT_BUFFER], arg ptr[in, kbase_hwcnt_reader_metadata])
+ioctl$KBASE_HWCNT_READER_SET_INTERVAL(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_SET_INTERVAL], arg const[0])
+ioctl$KBASE_HWCNT_READER_ENABLE_EVENT(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_ENABLE_EVENT], arg const[0])
+ioctl$KBASE_HWCNT_READER_DISABLE_EVENT(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_DISABLE_EVENT], arg const[0])
+ioctl$KBASE_HWCNT_READER_GET_API_VERSION(fd fd_hwcnt, cmd const[KBASE_HWCNT_READER_GET_API_VERSION], arg const[0])
+
+type base_atom_id int8
+
+base_jd_atom_v2 {
+	jc		int64
+	udata		base_jd_udata
+	extres_list	int64
+	nr_extres	int16
+	compat_core_req	int16
+	pre_dep		array[base_dependency, 2]
+	atom_number	base_atom_id
+	prio		flags[base_jd_prio, int8]
+	device_nr	int8
+	jobslot		int8
+	core_req	flags[base_jd_core_req, int32]
+	renderpass_id	int8
+	padding		array[int8, 7]
+}
+
+base_dependency {
+	atom_id		base_atom_id
+	dependency_type	flags[base_jd_dep_type, int8]
+}
+
+base_jd_udata {
+	blob	array[int64, 2]
+}
+
+kbase_ioctl_job_submit {
+	addr		ptr64[out, array[base_jd_atom_v2]]
+	nr_atoms	len[addr, int32]
+# sizeof(base_jd_atom_v2)
+	stride		const[56, int32]
+}
+
+kbase_ioctl_soft_event_update {
+	event		gpu_addr
+	new_status	int32
+	flags		const[0, int32]
+}
+
+kbase_ioctl_version_check {
+	major	int16
+	minor	int16
+}
+
+kbase_ioctl_set_flags {
+	create_flags	int32
+}
+
+kbase_ioctl_get_gpuprops {
+	buffer	ptr64[out, array[int8]]
+	size	len[buffer, int32]
+	flags	const[0, int32]
+}
+
+kbase_ioctl_mem_alloc_in {
+	va_pages	int64
+	commit_pages	int64
+	extent		int64
+	flags		int64
```

Comment:
Would this be flags[base_mem_alloc_flags, int64] type?

Reply:
Fixed.